pkg/backup/backup-session.go

```diff
@@ -0,0 +1,361 @@
+			err := backupOutput.HandleMetrics(&c.Metrics, nil)
+			if err != nil {
+				return err
+			}
+		}
+		// If output directory specified, then write the output in "output.json" file in the specified directory
+		if c.OutputDir != "" {
+			err := backupOutput.WriteOutput(filepath.Join(c.OutputDir))
+			if err != nil {
+				return err
+			}
+		}
+	}
+	//Update Backup Session and Repository status
+	//_, err = stash_v1beta1_util.UpdateRestoreSessionStatus()
+	o := status.UpdateStatusOptions{
+		Namespace:               c.Namespace,
+		BackupSession:           backupSession.Name,
+		Repository:              backupConfiguration.Spec.Repository.Name,
+		EnableStatusSubresource: apis.EnableStatusSubresource,
+	}
+	err = o.UpdateBackupStatus(&c.BackupOutput, c.StashClient.(*cs.Clientset))
+	if err != nil {
+		return err
+	}
+	fmt.Println("everything ok")
+	return nil
+}
+
+func IsPending(backupsession *v1beta1_api.BackupSession) bool {
+	if backupsession.Status.Phase == v1beta1_api.BackupSessionPending || backupsession.Status.Phase == "" {
+		return true
+	}
+	return false
+}
+
+func (c *Controllers) runBackupSessionController(backupConfiguration *v1beta1_api.BackupConfiguration, stopCh <-chan struct{}) error {
+	// start BackupSession watcher
+	err := c.initBackupSessionWatcher(backupConfiguration)
+	if err != nil {
+		return err
+	}
+
+	// start BackupSession informer
+	c.StashInformerFactory.Start(stopCh)
+	for _, v := range c.StashInformerFactory.WaitForCacheSync(stopCh) {
+		if !v {
+			runtime.HandleError(fmt.Errorf("timed out waiting for caches to sync"))
+			return nil
+		}
+	}
+	c.bsQueue.Run(stopCh)
+	return nil
+}
+
+func (c *Controllers) HandleBackupFailure(err error) error {
+	backupSession, err := c.StashClient.StashV1beta1().BackupSessions(c.Namespace).Get(c.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	// set backupSession phase "Failed"
+	_, err = stash_v1beta1_util.UpdateBackupSessionStatus(c.StashClient.StashV1beta1(), backupSession, func(status *v1beta1_api.BackupSessionStatus) *v1beta1_api.BackupSessionStatus {
+		status.Phase = v1beta1_api.BackupSessionFailed
+		return status
+	}, apis.EnableStatusSubresource)
+
+	// write failure event
+	c.writeBackupFailureEvent(backupSession, err)
+
+	// send prometheus metrics
+	if c.Metrics.Enabled {
+		restoreOutput := &restic.RestoreOutput{
+			SessionDuration: "0s",
+		}
+		return restoreOutput.HandleMetrics(&c.Metrics, err)
+	}
+	return nil
+}
+
+func (c *Controllers) writeBackupFailureEvent(backupSession *v1beta1_api.BackupSession, err error) {
+	// write failure event
+	ref, rerr := reference.GetReference(stash_scheme.Scheme, backupSession)
+	if rerr == nil {
+		eventer.CreateEventWithLog(
+			c.K8sClient,
+			controller.BackupSessionEventComponent,
+			ref,
+			core.EventTypeWarning,
+			eventer.EventReasonBackupSessionFailed,
+			fmt.Sprintf("Failed to restore. Reason: %v", err),
+		)
+	} else {
+		log.Errorf("Failed to restore. Reason: %v", err)
```

Comment:
Use `log.Errorf("Failed to write failure event. Reason: %v", rerr)`

Reply:
fixed